airflow/kubernetes/pod_generator.py

```diff
@@ -324,47 +325,36 @@ def add_sidecar(pod: k8s.V1Pod) -> k8s.V1Pod:
     @staticmethod
     def from_obj(obj) -> Optional[k8s.V1Pod]:
         """Converts to pod from obj"""
+
         if obj is None:
             return None
 
-        if isinstance(obj, PodGenerator):
-            return obj.gen_pod()
-
-        if not isinstance(obj, dict):
-            raise TypeError(
-                'Cannot convert a non-dictionary or non-PodGenerator '
-                'object into a KubernetesExecutorConfig')
+        k8s_legacy_object = obj.get("KubernetesExecutor", None)
+        k8s_object = obj.get("podOverride", None)
```

Comment:
```suggestion
        k8s_object = obj.get("pod_override", None)
```

I think using snake_case might be better, wdyt?

Reply:
@kaxil sure, I was trying to keep in line with the original key "KubernetesExecutor", but I'm fine with that to be more consistent with pythonic best practices.

Reply:
@kaxil fixed